src/utils/sovryn/contract-writer.ts

```diff
@@ -138,21 +139,26 @@ class ContractWriter {
       );
     } else {
       const { address, abi } = getContract(contractName);
+
       return this.sendByAddress(address, abi, methodName, args, options);
     }
   }
 
   public async sendByAddress(
     address: string,
-    abi: AbiItem[],
+    abi: AbiItem[] | AbiItem | any,
     methodName: string,
-    args: Array<any>,
+    args: any[],
     options: TransactionConfig = {},
   ): Promise<string | RevertInstructionError> {
+    let params = args;
+    if (args && args.length && typeof args[args.length - 1] === 'object') {
+      params = args.slice(0, -1);
+    }
     return new Promise<string | RevertInstructionError>(
       async (resolve, reject) => {
         const data = this.getCustomContract(address, abi)
-          .methods[methodName](...args)
+          .methods[methodName](...params)
```

Comment:
Not sure if this will not break other functionality, will need to make some tests.

Reply:
Looks like other dapp parts didn't broke, so should be ok